File: Second_Year/DAAA(Design_and_Analysis_of_Algorithms)/compare.java
import java.util.Scanner;

public class compare {
    public static void comparer(int a, int b) {
        int c = a-b;
        if (c > 0) {
            System.out.println(a + " > " + b);
        } else if (c < 0) {
            System.out.println(a + " < " + b);
        } else {
            System.out.println(a + " and " + b + " are equal");
        }
    }

    public static void main(String[] args) {
        System.out.print("Enter two numbers to compare: ");
        try (Scanner scan = new Scanner(System.in)) {
            int num1 = scan.nextInt();
            int num2 = scan.nextInt();
            comparer(num1, num2);
            scan.close();
        }
    }
}

File: Second_Year/DAAA(Design_and_Analysis_of_Algorithms)/compare.cpp
#include <iostream>
#include <limits>
using namespace std;

int compare(int a, int b) {
    int c = a-b;
    if(c < 0) {
        cout << b << " is larger" << endl;
    }
    else if (c > 0) {
        cout << a << " is larger" << endl;
    }
    else cout << "Numbers are equal";
}

int main() {
    int a, b;
    cout << "Enter 2 integers to compare them - ";
    while(!(cin >> a >> b)) {
        cout << "Please enter a valid integer - ";
        cin.clear();
        cin.ignore(numeric_limits<streamsize>::max(), '\n');
    }
    compare(a, b);
}